### PR TITLE
[mono] Community changes

### PR DESCRIPTION
**Pull request by stristr**
OK to merge.

Medium-length review.

Purpose of changes:

- Everything that needs to change in mono to support community Phase 2.

Summary of work (include Asana links if any):

- [x] Modify "replace summonables with default `0`" functionality (currently used with Lottie) to instead use default `1`, which matches edit mode and is overall safer (e.g. you can divide by 1).
- [x] Extend Core right-click menu to show a "Fork this component" option. We use the Fetch API (browser support is good enough, especially on Mac) wrapped in a try/catch to check with Inkstone if the project should be "forkable".
- [x] `@haiku/sdk-inkstone` tweaks to community API interfaces to match final functionality/needs.
- [x] New "HaikuStatic" bundle generated during publish. (I made the call not to experiment flag this because the exporter integration is known to be extremely fault-tolerant and has good test coverage for its fault tolerance.)
- [x] Adds support for `haiku://fork/:organizationName/:projectName`-type URLs in Creator, including the ability to pop a new/unauthed user straight into fork-land as soon as they sign in.

Regressions to look for:

- None expected, but post-merge this should all go through QA.

Notes for reviewers:

- This PR has significant Inkstone and community site dependencies that are being PR'ed asynchronously—specifically around forking functionality (including the check to see if the right-click menu should show the forking option at all and the actual forking functionality). This will make end-to-end testing difficult if you don't have a local inkstone setup. However, for safe merge, you only need to confirm that the app works correctly even without  any of the inkstone work deployed. In Core, this means: the request to "see if this project can be forked" should fail until that endpoint is actually deployed, so the "Fork this project" option shouldn't show up in the wild. In the app, this means: as long as you don't go to a `haiku://fork/…` type of URL, all the usual project init stuff is working fine.

Completed checkin tasks:

- [x] Did manual testing of interrelated functionality
- [x] Added measurement instrumentation (Mixpanel, etc.)
- [x] Expanded the [QA checklist](https://haiku.quip.com/rqwsAPsCGxqT)
- [x] Wrote an automated test covering new functionality

### CODE REVIEW

File: packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts
Comment (roperzh) [inline packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts]: Heads up, _just in case_ (I'm not saying that we should change it!), here's the [browser support](https://caniuse.com/#feat=fetch) for `fetch`

screen shot 2018-03-12 at 09 51 29

File: packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts
Comment (roperzh) [inline packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts]: nit: why don't just use a template literal for the whole string?

**Review by roperzh — APPROVED** [review]
LGTM! very nice!

File: packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts
Comment (tsortwehttam) [inline packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts]: Properties may also be objects, arrays, booleans, and `null`.

File: packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts
Comment (tsortwehttam) [inline packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts:43]: Just noting since I haven't traced this to code upstream, but it's possible that haiku/core (and possibly, by accident, some of our model utilities) attach properties to the bytecode that result in the object being circular, which would cause a crash at this line. (If need be, `ActiveComponent#getReifiedDecycledBytecode` can be called to get an object without cycles.)

File: packages/haiku-formats/src/exporters/injectables.ts
Comment (tsortwehttam) [inline packages/haiku-formats/src/exporters/injectables.ts:26]: This is a good idea. But (and keep in mind this may also be an issue in haiku/core), if somebody sets `rotation.[x|y|z]` to an expression,  a default value of 1 radians is quite different from 0 radians.

File: packages/haiku-plumbing/src/publish-hooks/saveExport.js
Comment (tsortwehttam) [inline packages/haiku-plumbing/src/publish-hooks/saveExport.js:5]: Heh, I guess nm my comment above

File: packages/haiku-creator/src/react/Creator.js
Comment (tsortwehttam) [inline packages/haiku-creator/src/react/Creator.js:1038]: Did you mean to early return here? (Also see line 1047 below where we may eventually want to add an early return.)

**Review by tsortwehttam — COMMENTED** [review]
Only one comment I think is necessary to address before merging, but look closely at my other comments which may be important (your call).

File: packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts
Comment (stristr) [reply packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts]: Yeah, I mentioned this without supporting data in PR description: "(browser support is good enough, especially on Mac)".

My approximate thinking is, if you're on mobile, you can't right-click. If you're on desktop and you're not using a supported browser on desktop, you're already not going to be impressed by Haiku enough to want to fork.

File: packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts
Comment (stristr) [reply packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts]: Ah yeah, I also thought about this way too much. I think mainly for readability, the surrounding code also broke lines, and I was going to for a light touch and didn't refactor. My IDE added the line breaks before `style` and `onMouseOut`, though, so I should take those out.

File: packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts
Comment (stristr) [reply packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts]: Hmmm…which? (Was going off `schema.ts`).

And for these, if we're short-circuiting an expression with the default return value of `1`, is that problematic?

File: packages/haiku-plumbing/src/publish-hooks/saveExport.js
Comment (stristr) [reply packages/haiku-plumbing/src/publish-hooks/saveExport.js:5]: :smile:

File: packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts
Comment (roperzh) [reply packages/@haiku/core/src/renderers/dom/createRightClickMenu.ts]: Sounds good! sorry for missing that bit in the PR description.

File: packages/haiku-formats/src/exporters/injectables.ts
Comment (stristr) [reply packages/haiku-formats/src/exporters/injectables.ts:26]: I believe this is identical to edit mode, but was having trouble finding the code…. @roperzh, can you confirm?

File: packages/haiku-creator/src/react/Creator.js
Comment (stristr) [reply packages/haiku-creator/src/react/Creator.js:1038]: Yes :smile:

File: packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts
Comment (tsortwehttam) [reply packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts]: JIT properties targeting (for example) <path> `d` and <polygon> `points` can take arrays describing the points data; we can accept a color object for colors; and `null` can signal no content for components that accept injected content. (This may not be relevant right now, but it will become a thing soon.) Maybe it's sufficient to just skip these.

Comment (stristr) [reply packages/haiku-formats/src/exporters/haikuStatic/haikuStaticExporter.ts]: Fixed [here](https://github.com/HaikuTeam/mono/commit/51ac015b). Also shored up Lottie export's ability to handle this.

**Comment by stristr** [comment]
Merging!

**Pull request merged**
